public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -365,7 +366,12 @@ export class PrometheusDatasource implements DataSourceApi<PromQuery> {
             };
 
             if (annotation.useValueForTime) {
-              event['time'] = Math.floor(parseFloat(value[1]));
+              const timestampValue = Math.floor(parseFloat(value[1]));
+              if (dupCheck[timestampValue]) {
+                continue;
+              }
+              dupCheck[timestampValue] = true;
+              event['time'] = timestampValue;
```

Comment:
why index lookup? shoulb be event.time = eventTime

Reply:
I don't remember why I did, anyway It should be `event.time` now.